gameboard.cpp

```diff
@@ -60,6 +62,23 @@ void GameBoard::setupPawns()
     }
 }
 
+void GameBoard::setupeRocks()
+{
+    std::array<size_t, 2> rocksRows {0, 7};
+    std::array<size_t, 2> rocksColumn {0, 7};
+
+    for (auto row : rocksRows) {
+        for (auto column : rocksColumn) {
+            BoardCellPosition position {row, column};
+            const CellIterator& cell = at(position);
+            cell->reset(new ChessPiece {row == 0 ? Side::Black : Side::White,
```

Comment:
Look - we have a duplicate code. Please, try to extract it to a new method. For example, piece color setting can be described as some predicate (lambda/functional object), which you'll pass as argument to a new method. And for pawns we can use accumulate method with array.